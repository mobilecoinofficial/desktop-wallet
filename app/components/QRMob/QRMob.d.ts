export interface QRMobProps {
  size: number;
  value: string;
}
