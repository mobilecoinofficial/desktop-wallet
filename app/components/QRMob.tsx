import React from 'react';
import type { FC } from 'react';

import { Box, makeStyles } from '@material-ui/core';
import PropTypes from 'prop-types';
import QRCode from 'qrcode.react';

import { CircleStrokeMOBIcon } from './icons';

interface QRMobProps {
  size: number;
  value: string;
}

const useStyles = makeStyles(() => ({
  container: {
    height: '296px',
    position: 'relative',
    width: '296px',
  },
  icon: {
    alignItems: 'center',
    display: 'flex',
    height: '100%',
    justifyContent: 'center',
    left: 0,
    position: 'absolute',
    top: 0,
    width: '100%',
    zIndex: 1000,
  },
  qr: {
    height: '100%',
    left: 6,
    position: 'absolute',
    top: 6,
    width: '100%',
  },
  root: {},
}));

const QRMob: FC<QRMobProps> = (props) => {
  const classes = useStyles();

  return (
    <Box className={classes.container}>
      <QRCode className={classes.qr} includeMargin level="H" {...props} />
      <Box className={classes.icon}>
        <CircleStrokeMOBIcon color="black" height={64} width={64} />
      </Box>
    </Box>
  );
};

QRMob.propTypes = {
  size: PropTypes.number.isRequired,
  value: PropTypes.string.isRequired,
};

export default QRMob;
