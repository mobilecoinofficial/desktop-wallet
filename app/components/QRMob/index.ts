import { QRMob } from './QRMob.view';

export default QRMob;
export { QRMob };
