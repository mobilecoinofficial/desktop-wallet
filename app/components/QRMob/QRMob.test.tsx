import React from 'react';

import 'jest-canvas-mock';
import { render } from '@testing-library/react';

import { QRMob } from './QRMob.view';

describe('QRMob', () => {
  test('renders to screen when called', () => {
    const { asFragment } = render(<QRMob size={280} value="mob:///b58/b58code" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
