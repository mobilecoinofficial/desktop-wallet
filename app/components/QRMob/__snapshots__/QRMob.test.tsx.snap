// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRMob renders to screen when called 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-5 makeStyles-container-1"
  >
    <canvas
      class="makeStyles-qr-3"
      height="280"
      style="height: 280px; width: 280px;"
      width="280"
    />
    <div
      class="MuiBox-root MuiBox-root-6 makeStyles-icon-2"
    >
      <svg
        fill="none"
        height="64"
        viewBox="0 0 43 43"
        width="64"
      >
        <circle
          cx="21.5"
          cy="21.5"
          fill="#fff"
          r="21"
          stroke="black"
        />
        <path
          d="M21.5 40C11.299 40 3 31.7 3 21.5S11.299 3 21.5 3 40 11.3 40 21.5C40 31.702 31.701 40 21.5 40zm8.957-22.37a2.41 2.41 0 01-4.346 1.438H6.064a15.741 15.741 0 00-.188 2.433c0 8.615 7.008 15.624 15.624 15.624 6.309 0 11.757-3.76 14.22-9.154H17.267a2.41 2.41 0 11-.08-2.875v-.001h19.521c.273-1.155.418-2.358.418-3.594 0-8.616-7.01-15.625-15.625-15.625-6.753 0-12.519 4.306-14.696 10.316H26.11a2.41 2.41 0 014.346 1.438z"
          fill="black"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
